cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -118,6 +118,7 @@ def self.config(env)
             ) +
             # TODO: Collapse these paths into /private to simplify Pegasus caching config.
             %w(
+              /amazon-future-engineer*
```

Comment:
What exactly does this do?

Reply:
This is a change to our Varnish configuration which makes `/amazon-future-engineer` behave like a session-specific page, which seemed appropriate if we're requiring login to view it.

Specifically, it lets the following cookies pass through the Varnish layer when loading this page:
```
language_
pm
_user_type
_limit_project_types
_assumed_identity
hour_of_code
progress
lines
scripts
videos_seen
callouts_seen
rack.session
remember_user_token
__profilin
_learn_session
storage_id
```
...and the `Accept-Language` and `CloudFront-Viewer-Country` headers.

If we didn't add this line, the pegasus default configuration would apply:

https://github.com/code-dot-org/code-dot-org/blob/5c130e8639fa440da6466d6fd998ea8383112cc8/cookbooks/cdo-varnish/libraries/http_cache.rb#L140-L144

...which would only allow the cookies:

```
language_
pm
_user_type
_limit_project_types
_assumed_identity
```

And now I'm seeing what you're saying about using the `_user_type` cookie for this.  Maybe I should update the `authentication_required!` helper to check the `_user_type` cookie instead of depending on `_learn_session`.

Reply:
Update: I've opted out of using the `_user_type` cookie for this because it's fairly easy to fake from the client.  I think it's reasonable to do less caching for this page given its intentionally limited audience.